sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/ResourceBase.java

```diff
@@ -181,4 +157,37 @@ protected Response fetchRows(
     }
     return resultRows;
   }
+
+  // // // Helper methods for input validation
+
+  protected static void requireNonEmptyKeyspace(String keyspaceName) {
+    if (isStringEmpty(keyspaceName)) {
+      throw new WebApplicationException(
+          "keyspaceName must be provided", Response.Status.BAD_REQUEST);
+    }
+  }
+
+  protected static void requireNonEmptyKeyspaceAndTable(String keyspaceName, String tableName) {
+    requireNonEmptyKeyspace(keyspaceName);
+    if (isStringEmpty(tableName)) {
+      throw new WebApplicationException("table name must be provided", Response.Status.BAD_REQUEST);
+    }
+  }
+
+  protected static final boolean isStringEmpty(String str) {
+    return (str == null) || str.isEmpty();
+  }
+
+  protected static byte[] decodeBase64(String base64encoded) {
+    // TODO: error handling
+    return Base64.getDecoder().decode(base64encoded);
+  }
+
+  // // // Helper methods for JAX-RS response construction
+
+  protected static Response restServiceError(Response.Status httpStatus, String failMessage) {
```

Comment:
is this needed anymore?

Reply:
It is used for explicit failure cases where handling locally differs from global exception handlers; for example to give more specific failure message (but sometimes different HTTP Code as well).
Older usages were removed by Olivier in cases where global handling works well, which was about all of them.